go/libraries/doltcore/sqle/dfunctions/dolt_merge.go

```diff
@@ -133,11 +143,10 @@ func DoDoltMerge(ctx *sql.Context, args []string) (int, int, error) {
 // mergeIntoWorkingSet encapsulates server merge logic, switching between
 // fast-forward, no fast-forward, merge commit, and merging into working set.
 // Returns a new WorkingSet, whether there were merge conflicts, and whether a
-// fast-forward was performed. This currently persists merge commits in the
-// database, but expects the caller to update the working set.
+// fast-forward was performed. This commits the working set if merge is successful and
+// -no-commit flag is not defined and autocommit is on
 // TODO FF merging commit with constraint violations requires `constraint verify`
-func mergeIntoWorkingSet(ctx *sql.Context, sess *dsess.DoltSession, roots doltdb.Roots, ws *doltdb.WorkingSet, dbName string, spec *merge.MergeSpec) (*doltdb.WorkingSet, int, int, error) {
-
+func mergeIntoWorkingSet(ctx *sql.Context, sess *dsess.DoltSession, roots doltdb.Roots, ws *doltdb.WorkingSet, dbName string, spec *merge.MergeSpec, noCommit bool, msg string) (*doltdb.WorkingSet, int, int, error) {
```

Comment:
This function is a good example of a nice encapsulations cope for a function – it handles all the merge logic and commits if it needs to be done. The caller has a really nice, clear, easy to understand interface to work with. This version is a good one to contrast with the earlier CLI/command version, from a caller perspective. This one takes care of everything and makes it really easy for the caller. 

However... we should change the name now 😉  `mergeIntoWorkingSet` is true... but it doesn't tell the full story. Most of the time this will create a commit, too, so workingset is a little misleading. `mergeAndCommit` or even just `performMerge` might be a little more intuitive of a name.